test/demo_client.py

```diff
@@ -11,24 +11,24 @@
 import blackduck
 
 def vulns_in_all_project_versions_components(bd):
-    for project in bd.get_resource(name='projects'):
-        for version in bd.get_resource(project, 'versions'):
-            for component in bd.get_resource(version, 'components'):
-                for vulnerability in bd.get_resource(component, 'vulnerabilities'):
+    for project in bd.get_resource('projects'):
+        for version in bd.get_resource('versions', project):
```

Comment:
Note the switch of 'name' and 'parent' resource. I've added Type checking to print out a meaningful error if the developer gets them mixed up.

Reply:
If desired to have them the other way around we could use named parameters for both i.e. get_resource(parent=None, name=None, ...).

Reply:
That works - certainly easier to use positionals than named arguments.